File: engine/engine-model/src/main/scala/tools/packet/foreigneffect/ShowForeignEffect.scala
package tools.packet.foreigneffect

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowForeignEffect(private var _characterId: Int, private var _effect: Int) extends PacketInput {
  def characterId: Int = _characterId

  def effect: Int = _effect

  override def opcode(): SendOpcode = SendOpcode.SHOW_FOREIGN_EFFECT
}

File: engine/engine-model/src/main/java/tools/packet/foreigneffect/ShowForeignEffect.java
package tools.packet.foreigneffect;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowForeignEffect(Integer characterId, Integer effect) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SHOW_FOREIGN_EFFECT;
   }
}